app/src/main/java/com/example/lastfm/ServiceGenerator.java

```diff
@@ -0,0 +1,60 @@
+package com.example.lastfm;
+
+import android.util.Log;
+import android.widget.Toast;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+import java.util.List;
+
+import okhttp3.OkHttpClient;
+import okhttp3.logging.HttpLoggingInterceptor;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+public class ServiceGenerator {
+
+    private static final String TAG = "Retrofit";
+
+    private static final String BASE_URL = "http://ws.audioscrobbler.com/2.0/";
+
+    private static Gson gson = new GsonBuilder()
+            .setLenient()
+            .create();
+
+    private static OkHttpClient.Builder httpClient =
+            new OkHttpClient.Builder();
+
+    private  static Retrofit.Builder builder = new Retrofit.Builder()
+            .baseUrl(BASE_URL)
+            .addConverterFactory(
+                    GsonConverterFactory.create(gson)
+            );
+
+    private static HttpLoggingInterceptor logging =
+            new HttpLoggingInterceptor()
+                    .setLevel(HttpLoggingInterceptor.Level.BODY);
+
+
+    private static Retrofit retrofit = builder.build();
+
+    public static <S> S createService(
+            Class<S> serviceClass) {
```

Comment:
The single parameter of the function must not be  moved to the next line

Reply:
Moved to top